### PR TITLE
Sample datasets; reference for book chapter; misc docs

### PR DESCRIPTION
**Pull request by uellue**
Update the sample dataset section. Thx @twentyse7en and @sayandip18 for the build up!

Additional smaller fixes such as updating the book chapter BibTeX entry.

## Contributor Checklist:

* [x] I have added myself to [the creators.json file](https://github.com/LiberTEM/LiberTEM/blob/master/packaging/creators.json)
* [x] I have added [a changelog entry](https://github.com/LiberTEM/LiberTEM/tree/master/docs/source/changelog) for my contribution
* [x] I have added/updated documentation for all user-facing changes

### CODE REVIEW

**Comment by uellue** [comment]
I can't reproduce the missing BibTeXt entry error locally. Closing and reopening to see if it helps.

**Comment by uellue** [comment]
@sk1p can you reproduce it locally? I am out of ideas.

**Comment by sk1p** [comment]
I couldn't when I tried before - I suspect something is caching bibtex info somewhere? Maybe you can try to run docs-check in a clean environment, for example a docker container?

**Comment by sayandip18** [comment]
I think the citation for the bullseyeprobe should be on line 15 instead of 16.  The citation for Electron Bessel Beam is on the next line to the link.

**Comment by sayandip18** [comment]
``:cite:`ophus_colin_2019_3592520,Zeltmann2019``  and ``:cite:`giulio_guzzinati_2019_2566137,Guzzinati2019`` should be links, right?

**Comment by sayandip18** [comment]
If they are links, then we should have some kind of `` . . _`name`:  https://thelink.com`` thing at the end of the document. I think that's why it's failing.

**Comment by sk1p** [comment]
@sayandip18 can you reproduce this error if you clone @uellue 's fork and check out the `docs` branch? If so, you can try if your suggestion fixes the error

File: docs/source/sample_datasets.rst [hunk 1]
```diff
@@ -1,44 +1,53 @@
...
-.. _Bullseye probe: https://zenodo.org/record/3592520#.XmdNN3DhXIU
-.. _Circular probe: https://zenodo.org/record/3592520#.XmdNN3DhXIU  
-.. _Electron Bessel beam diffraction pattern: https://zenodo.org/record/2566137#.XmdNQnDhXIU
-.. _Python API: https://libertem.github.io/LiberTEM/api.html
\ No newline at end of file
+Alternatively, you can enter the parameters (scan_size, dtype, detector_size)
+directly into the load dialog of the GUI.
+
+.. _`Bullseye and circular probe diffraction`: https://zenodo.org/record/3592520
+.. _`Electron Bessel beam diffraction`: https://zenodo.org/record/2566137
```
Comment (sayandip18) [inline docs/source/sample_datasets.rst:53]:
```suggestion
.. _`Electron Bessel beam diffraction`: https://zenodo.org/record/2566137
```

**Comment by uellue** [comment]
> I couldn't when I tried before - I suspect something is caching bibtex info somewhere? Maybe you can try to run docs-check in a clean environment, for example a docker container?

Good call! I could reproduce it in a minimal clean docker environment.

**Comment by sayandip18** [comment]
> > I couldn't when I tried before - I suspect something is caching bibtex info somewhere? Maybe you can try to run docs-check in a clean environment, for example a docker container?
> 
> Good call! I could reproduce it in a minimal clean docker environment.

Oops! I accidentally pressed the wrong button

File: docs/source/sample_datasets.rst (hunk 1 above)
Comment (sayandip18) [inline docs/source/sample_datasets.rst:53]:
```suggestion
.. _`Electron Bessel beam diffraction`: https://zenodo.org/record/2566137
.. bibliography:: libertem-references.bib
```

**Comment by sk1p** [comment]
> Good call! I could reproduce it in a minimal clean docker environment

:+1: - maybe `strace` can help to track this one down

File: docs/source/sample_datasets.rst (hunk 1 above)
Comment (uellue) [reply docs/source/sample_datasets.rst:53]: So did you reproduce the error on your system and confirm that this fixes it?

Comment (uellue) [reply docs/source/sample_datasets.rst:53]: On my system I get an error message when I have two bibliographies.

Comment (sayandip18) [reply docs/source/sample_datasets.rst:53]: > So did you reproduce the error on your system and confirm that this fixes it?

No I haven't confirmed that. I saw [this](https://build-me-the-docs-please.readthedocs.io/en/latest/Using_Sphinx/UsingBibTeXCitationsInSphinx.html#example) in the Sphinx documentation and suggested the changes.



> On my system I get an error message when I have two bibliographies.

How did you test this in your system?

**Comment by uellue** [comment]
That's an interesting one: The `.. bibliography:: references-libertem.bib` has to be in the last file that sphinx reads. I now have a file `zzz-bibliography.rst` instead of having the bibliography in index.rst, and it works.

File: docs/source/sample_datasets.rst (hunk 1 above)
Comment (uellue) [reply docs/source/sample_datasets.rst:53]: `tox -e docs-check` https://libertem.github.io/LiberTEM/contributing.html#building-the-documentation

**Comment by sk1p** [comment]
Looks like this took care of the bibtex caching issue, well done. PR looks good, thanks!

**Pull request merged**